server/src/main/java/org/elasticsearch/index/fieldvisitor/FieldsVisitor.java

```diff
@@ -85,13 +85,13 @@ public Status needsField(FieldInfo fieldInfo) {
         // All these fields are single-valued so we can stop when the set is
         // empty
         return requiredFields.isEmpty()
-                ? Status.STOP
-                : Status.NO;
+            ? Status.STOP
+            : Status.NO;
     }
 
-    public final void postProcess(Function<String, MappedFieldType> fieldTypeLookup, @Nullable DocumentMapper mapper) {
-        if (mapper != null) {
-            type = mapper.type();
+    public final void postProcess(Function<String, MappedFieldType> fieldTypeLookup, @Nullable Supplier<String> typeSupplier) {
+        if (typeSupplier != null) {
+            type = typeSupplier.get();
```

Comment:
the supplier feels weird, yet I believe we need to distinguish between a null document mapper and a null type, hence simply providing the type as an argument did not feel right to me. Happy to be proven wrong.

Reply:
The only time you would have a null type is if no mappings were configured; in which case there would be no documents in the index, and FieldsVisitor would not be able to load any data.  So I think you can just replace this with a straight String - it might even be worth adding an assertion that the type is not null here?

Reply:
ok will do thanks for the feedback